File: python/meteroid/models/custom_property_definition.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .custom_property_definition_id import CustomPropertyDefinitionId
from .custom_property_entity_type import CustomPropertyEntityType
from .custom_property_type import CustomPropertyType
from .property_config import PropertyConfig


@dataclasses.dataclass
class CustomPropertyDefinition(BaseModel):
    archived: bool

    config: PropertyConfig

    display_order: int

    entity_type: CustomPropertyEntityType

    id: CustomPropertyDefinitionId

    key: str

    name: str

    property_type: CustomPropertyType

    required: bool

    default_value: t.Optional[t.Any] = None

    description: t.Optional[str] = None

File: go/custom_property_definition.go
// this file is @generated
package meteroid

import "encoding/json"

type CustomPropertyDefinition struct {
	Archived bool `json:"archived"`

	Config PropertyConfig `json:"config"`

	DefaultValue json.RawMessage `json:"default_value,omitempty"`

	Description *string `json:"description,omitempty"`

	DisplayOrder int32 `json:"display_order"`

	EntityType CustomPropertyEntityType `json:"entity_type"`

	Id CustomPropertyDefinitionId `json:"id"`

	Key string `json:"key"`

	Name string `json:"name"`

	PropertyType CustomPropertyType `json:"property_type"`

	Required bool `json:"required"`
}
